### PR TITLE
Announced Extended now sends a report for station traits + minor punctuation cleanup

### PR DESCRIPTION
**Pull request by MrMelbert**
## About The Pull Request

- Fixes two minor punctuation errors in station traits.
- Announced greenshifts now send a command report so players can figure out what station traits may be affecting the station.
- Changed some code in game_mode and dynamic `send_intercept()` to be their own procs to reduce copy+paste.

## Why It's Good For The Game

Announced greenshifts aren't run normally / are admin only but if they happen to be ran, and the station has traits, the crew should be able to know what they're affected with.

## Changelog
:cl: Melbert
qol: Announced Extended shifts now send a command report listing station traits.
spellcheck: Fixed some punctuation in station trait reports.
code: Reduced some copy+paste in gamemode intercepts
/:cl:

### CODE REVIEW

File: code/game/gamemodes/game_mode.dm [hunk 1]
```diff
@@ -293,25 +293,33 @@
...
-		for(var/datum/station_goal/G in station_goals)
-			G.on_report()
-			intercepttext += G.get_report()
-
-	if(SSstation.station_traits.len)
-		intercepttext += "<hr><b>Identified shift divergencies:</b>"
-		for(var/i in SSstation.station_traits)
-			var/datum/station_trait/station_trait_iterator = i
-			if(!station_trait_iterator.show_in_report)
-				return
```
Comment (MrMelbert) [inline code/game/gamemodes/game_mode.dm:307]: Turns out for normal gamemodes, the fix for station traits blocking reports was never mirrored over 🥴

File: code/game/gamemodes/game_mode.dm [hunk 2]
```diff
@@ -293,25 +293,33 @@
...
 	if(GLOB.security_level < SEC_LEVEL_BLUE)
 		set_security_level(SEC_LEVEL_BLUE)
 
+/datum/game_mode/proc/generate_station_goal_report()
+	if(!station_goals.len)
+		return
+	. = "<hr><b>Special Orders for [station_name()]:</b><BR>"
+	for(var/datum/station_goal/G in station_goals)
+		G.on_report()
+		. += G.get_report()
```
Comment (Mothblocks) [inline code/game/gamemodes/game_mode.dm:310]:
```suggestion
	for(var/datum/station_goal/station_goal in station_goals)
		station_goal.on_report()
		. += station_goal.get_report()
```

File: code/game/gamemodes/game_mode.dm [hunk 3]
```diff
@@ -293,25 +293,33 @@
...
+
+/datum/game_mode/proc/generate_station_trait_report()
+	if(!SSstation.station_traits.len)
+		return
+	. = "<hr><b>Identified shift divergencies:</b><BR>"
+	for(var/i in SSstation.station_traits)
+		var/datum/station_trait/station_trait_iterator = i
+		if(!station_trait_iterator.show_in_report)
+			continue
+		. += "[station_trait_iterator.get_report()]<BR>"
```
Comment (Mothblocks) [inline code/game/gamemodes/game_mode.dm:321]:
```suggestion
	for(var/datum/station_trait/station_trait as anything in SSstation.station_traits)
		if(!station_trait.show_in_report)
			continue
		. += "[station_trait.get_report()]<BR>"
```

File: code/game/gamemodes/game_mode.dm [hunk 4]
```diff
@@ -293,25 +293,33 @@
...
-			intercepttext += station_trait_iterator.get_report()
+	intercepttext += generate_station_goal_report()
+	intercepttext += generate_station_trait_report()
 
 	print_command_report(intercepttext, "Central Command Status Summary", announce=FALSE)
 	priority_announce("A summary has been copied and printed to all communications consoles.", "Enemy communication intercepted. Security level elevated.", ANNOUNCER_INTERCEPT)
 	if(GLOB.security_level < SEC_LEVEL_BLUE)
 		set_security_level(SEC_LEVEL_BLUE)
 
+/datum/game_mode/proc/generate_station_goal_report()
```
Comment (Mothblocks) [inline code/game/gamemodes/game_mode.dm:304]: Document.

**Review by Mothblocks — CHANGES_REQUESTED** [review]

**Pull request merged**